### PR TITLE
[linalg] Add handling for leadin and trailing size-1 dims in ViewOp

### PR DESCRIPTION
**Pull request by ramiro050**
After trying to generalize https://github.com/llvm/torch-mlir/pull/2474 to handle leading size-1 dims, it became clear that the approach used there was not the right one. Therefore, this PR:

- Reverts https://github.com/llvm/torch-mlir/pull/2474
- Adds to the lowering of `aten.view` handling for the following cases:
  - `(..., a.size(i))` -> `(..., a.size(i), 1, ..., 1)`
  - `(..., a.size(i), 1, ..., 1)` -> `(..., a.size(i))`
  - `(a.size(i), ...)` -> `(1, ..., 1, a.size(i), ...)`
  - `(1, ..., 1, a.size(i), ...)` -> `(a.size(i), ...)`

I recommend reviewing commits separately.

### CODE REVIEW

File: python/torch_mlir_e2e_test/test_suite/reshape_like.py [hunk 1]
```diff
@@ -672,6 +672,74 @@ def forward(self, a):
...
+        ([1, 1, 1, -1], torch.float32, True),
+    ])
+
+    def forward(self, a):
+        return a.view(a.size(3))
+
+@register_test_case(module_factory=lambda: ViewSizeDimLedByCollapsedOnesModule())
+def ViewSizeDimLedByCollapsedOnesModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(1, 1, 1, 128))
+
```
Comment (qedawkins) [inline python/torch_mlir_e2e_test/test_suite/reshape_like.py:742]: It might be worth adding a test for both at the same time, e.g. `?` -> `1, 1, ?, 1, 1`

**Review by qedawkins — APPROVED** [review]
Thanks for the fix. Agree with the revert and the new fix looks good to me.

File: python/torch_mlir_e2e_test/test_suite/reshape_like.py (hunk 1 above)
Comment (ramiro050) [reply python/torch_mlir_e2e_test/test_suite/reshape_like.py:742]: Good idea. Added!

**Review by brucekimrokcmu — APPROVED** [review]
LGTM

**Review by qedawkins — APPROVED** [review]

**Review by dan-garvey — APPROVED** [review]
Thanks, I was dragging my feet a little as I had some other kinks to work out but this looks like it will work

**Pull request merged**